app/src/modules/extension/mv3RuleParser/parseReplaceRule.ts

```diff
@@ -10,18 +9,20 @@ const parseReplaceRule = (rule: ReplaceRule): ExtensionRule[] => {
 
   rule.pairs.forEach((rulePair, pairIndex) => {
     if (!rulePair.source.value) {
-      extensionRules.push({
+      const ruleCondition = {
         condition: {
           regexFilter: `(.*)${rulePair.from}(.*)`,
-          excludedRequestDomains: BLACKLISTED_DOMAINS,
+          ...parseConditionFromSource(rulePair.source),
         },
         action: {
           type: RuleActionType.REDIRECT,
           redirect: {
             regexSubstitution: `\\1${rulePair.to}\\2`,
           },
         },
-      });
+      };
+      delete ruleCondition.condition.urlFilter;
```

Comment:
Why only in replace rule? Should also be in redirect and query params rule 🤔

Anyway, this should be handled in different place as this is a migration. Check other comment

Reply:
This is not related to migration. Its a fix: DNR source filter conditions were not generated when replace rule was applied to all the URLs